packages/venia-ui/lib/util/apolloCache.js

```diff
@@ -63,3 +65,223 @@ export const cacheKeyFromType = object => {
             return defaultDataIdFromObject(object);
     }
 };
+
+const productKeyFieldFunction = object => {
+    return object.url_key
+        ? `${MagentoGraphQLTypes.ProductInterface}:${object.url_key}`
+        : defaultDataIdFromObject(object);
+};
+
+/**
+ * Replaces the deprecated cacheKeyFromType.
+ */
+export const TYPE_POLICIES = {
```

Comment:
>  it seems that empty lines have been avoided throughout this object

Not intentional. I worked on this object in pieces as I debugged all the various broken queries/mutations. Over time it grew into what it is, but doesn't have to be it's _final form_.

> Could there be a way we could move type policies into a file (or files) where things might be broken out a little more? 

Yes! You might have seen my experiment with the custom `product` query. You can dynamically add type policies even more dynamically than with resolvers. `apolloClient.cache.policies.addTypePolicies(typePolicies)` should do the trick, but does come with a caveat: Types don't "belong" to a component. You can create multiple components that query and mutate different types, so where should we define the policies and where should we call `addTypePolicies`? It made sense to dynamically create and add the custom query in `product.gql.js` since the product component is where we call the query. I'm not sure where we would add type policies for the different pieces of `Cart`.